core/src/main/java/org/prevayler/implementation/TransactionWithQueryCapsule.java

```diff
@@ -19,9 +19,9 @@ public TransactionWithQueryCapsule(byte[] serialized) {
     super(serialized);
   }
 
-  protected void justExecute(Object transaction, Object prevalentSystem, Date executionTime) {
+  protected void justExecute(TransactionWithQuery<? super P, R> transaction, P prevalentSystem, Date executionTime) {
     try {
-      _queryResult = ((TransactionWithQuery<P, R>) transaction).executeAndQuery((P) prevalentSystem, executionTime);
+      _queryResult = transaction.executeAndQuery((P) prevalentSystem, executionTime);
```

Comment:
I guess the `(P)` isn't necessary here anymore.

Reply:
Sure thing. There are some more spots I missed so far; please expect further updates within my clone.

Reply:
Please see the recent pushes on my cloned repo.